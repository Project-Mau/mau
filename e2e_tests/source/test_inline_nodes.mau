= Test inline nodes

This document contains the standard inline nodes that can appear in a paragraph.

This is a normal paragraph that is followed by an empty line.

This is a paragraph split in two lines.
The missing empty line makes the two line belong to the same paragraph.

This paragraph contains `verbatim` text and the supported styles: _underscore_, *star*, ~tilde~ and ^caret^.

Here we use [class]("classes", "class1,class2").

A paragraph with an inline image [image]("https://www.thedigitalcatonline.com/images/global/favicon.jpg").

Links like there was no [link]("https://www.thedigitalcatonline.com", "tomorrow"). Also without text: [link]("https://www.thedigitalcatonline.com").

You can use the macro `mailto` to format email links. If you need to know more write me at [mailto]("info@example.com").
