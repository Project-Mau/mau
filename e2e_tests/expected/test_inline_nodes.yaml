data:
  args: []
  content:
  - data:
      anchor: test-inline-nodes-378b
      args: []
      kwargs: {}
      level: 1
      tags: []
      type: header
      value: Test inline nodes
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: This document contains the standard inline nodes that can appear
                in a paragraph.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: This is a normal paragraph that is followed by an empty line.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: This is a paragraph split in two lines. The missing empty line
                makes the two line belong to the same paragraph.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'This paragraph contains '
          - data:
              type: verbatim
              value: verbatim
          - data:
              type: text
              value: ' text and the supported styles: '
          - data:
              content:
                data:
                  content:
                  - data:
                      type: text
                      value: underscore
                  type: sentence
              type: style
              value: underscore
          - data:
              type: text
              value: ', '
          - data:
              content:
                data:
                  content:
                  - data:
                      type: text
                      value: star
                  type: sentence
              type: style
              value: star
          - data:
              type: text
              value: ', '
          - data:
              content:
                data:
                  content:
                  - data:
                      type: text
                      value: tilde
                  type: sentence
              type: style
              value: tilde
          - data:
              type: text
              value: ' and '
          - data:
              content:
                data:
                  content:
                  - data:
                      type: text
                      value: caret
                  type: sentence
              type: style
              value: caret
          - data:
              type: text
              value: .
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'Here we use '
          - data:
              classes:
              - class1
              - class2
              content:
                data:
                  content:
                  - data:
                      type: text
                      value: classes
                  type: sentence
              type: class
          - data:
              type: text
              value: .
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'A paragraph with an inline image '
          - data:
              alt_text: null
              height: null
              type: image
              uri: https://www.thedigitalcatonline.com/images/global/favicon.jpg
              width: null
          - data:
              type: text
              value: .
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'Links like there was no '
          - data:
              target: https://www.thedigitalcatonline.com
              text: tomorrow
              type: link
          - data:
              type: text
              value: '. Also without text: '
          - data:
              target: https://www.thedigitalcatonline.com
              text: https://www.thedigitalcatonline.com
              type: link
          - data:
              type: text
              value: .
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'You can use the macro '
          - data:
              type: verbatim
              value: mailto
          - data:
              type: text
              value: ' to format email links. If you need to know more write me at '
          - data:
              target: mailto:info@example.com
              text: info@example.com
              type: link
          - data:
              type: text
              value: .
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  kwargs: {}
  tags: []
  type: document

